includes/class-wc-webhook.php

```diff
@@ -587,6 +610,15 @@ public function log_delivery( $delivery_id, $request, $response, $duration ) {
 	 * @since 2.2.0
 	 */
 	private function failed_delivery() {
+		if ( $this->should_retry() ) {
+			// Retry.
+			add_action( 'woocommerce_webhook_delivery', array( $this, 'retry_delivery' ), 10, 5 );
```

Comment:
I worry that hooking to `woocomerce_webhook_delivery` adds some confusion to the retry process, what do you think? (It's also worth noting this isn't removed and every webhook executed after a failure will be retried!)

If having access to the variables passed in `WC_Webhook::delivery()` is a priority, perhaps we could put this in `WC_Webhook::log_delivery()` and pass the `$arg` through to `WC_Webhook::retry_delivery()`? We could also then make the latter private and avoid extending the public API of this class. What do you think?

Reply:
Yeah, this was my biggest concern with porting my previous "hack" over -- I'll implement this today.